Source/JavaScriptCore/runtime/MathObject.cpp

```diff
@@ -450,20 +450,33 @@ JSC_DEFINE_HOST_FUNCTION(mathProtoFuncSumPrecise, (JSGlobalObject* globalObject,
     if (iterable.isUndefinedOrNull())
         return throwVMTypeError(globalObject, scope, "Math.sumPrecise requires first argument not be null or undefined"_s);
 
-    PreciseSum sum;
+    std::optional<uint64_t> length = std::nullopt;
+    if (JSC::isArray(globalObject, iterable)) {
+        length = toLength(globalObject, iterable.toObject(globalObject));
```

Comment:
If we find it is `JSArray`, you should use `jsCast<JSArray*>(iterable)` instead of `toObject`.

Reply:
So, we should use `jsCast<JSArray*>(iterable)->length()` instead of `toLength`.